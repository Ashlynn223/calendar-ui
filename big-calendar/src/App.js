import { Calendar, momentLocalizer } from "react-big-calendar";
import moment from "moment";
import "./App.css";
import "react-big-calendar/lib/css/react-big-calendar.css";
import { useState } from "react";

const localizer = momentLocalizer(moment);
const currentHour = new Date().getHours();
const nextHour = currentHour + 1;
const currentMinutes = new Date().getMinutes();
const formatToHourAndMinutes = (time) => time + ":00:00.000";
const defaultStartTime =
  currentMinutes > 0
    ? formatToHourAndMinutes(nextHour)
    : formatToHourAndMinutes(currentHour);
const defaultEndTime = formatToHourAndMinutes(nextHour + 1);

function padTo2Digits(num) {
  return num.toString().padStart(2, "0");
}

function formatDate(date) {
  return [
    date.getFullYear(),
    padTo2Digits(date.getMonth() + 1),
    padTo2Digits(date.getDate()),
  ].join("-");
}

const formatAMPM = (timeString) => {
  const date = timeString.substring(0, 5);
  let hours = date.substring(0, 2);
  let minutes = date.substring(3, 5);
  let ampm = hours >= 12 ? "pm" : "am";
  hours = hours % 12;
  hours = hours ? hours : 12;
  let strTime = hours + ":" + minutes + " " + ampm;
  return strTime;
};

function App() {
  const [startDate, setStartDate] = useState(formatDate(new Date())); // may set this to a Date object later.
  const [endDate, setEndDate] = useState(formatDate(new Date()));
  const [startTime, setStartTime] = useState(defaultStartTime);
  const [endTime, setEndTime] = useState(defaultEndTime);
  const [title, setTitle] = useState("");
  const [events, setEvents] = useState([]);

  const handleSubmit = (e) => {
    e.preventDefault();
    setEvents([
      {
        title: `${formatAMPM(startTime)} ${title}`,
        start: new Date(`${startDate}T${startTime}`),
        end: new Date(`${endDate}T${endTime}`),
      },
    ]);
  };

  return (
    <div className="App">
      <div>
        <Calendar
          localizer={localizer}
          events={events}
          startAccessor="start"
          endAccessor="end"
          style={{ height: 500 }}
        />
      </div>
      <label>Title</label>
      <input
        type="text"
        onChange={(e) => {
          setTitle(e.target.value);
        }}
        value={title}
      />

      <label>Start Date</label>
      <input
        type="date"
        onChange={(e) => {
          setStartDate(e.target.value);
        }}
        value={startDate}
      />
      <label>Start Time</label>
      <input
        type="time"
        onChange={(e) => {
          setStartTime(e.target.value);
        }}
        value={startTime}
      />
      <label>End Date</label>
      <input
        type="date"
        onChange={(e) => {
          setEndDate(e.target.value);
        }}
        value={endDate}
      />
      <label>End Time</label>
      <input
        type="time"
        onChange={(e) => {
          setEndTime(e.target.value);
        }}
        value={endTime}
      />
      <button onClick={handleSubmit}>Create Event</button>
    </div>
  );
}

export default App;
