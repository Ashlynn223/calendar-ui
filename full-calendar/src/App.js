import React, { useState } from "react";
import FullCalendar from "@fullcalendar/react"; // must go before plugins
import dayGridPlugin from "@fullcalendar/daygrid"; // a plugin!
import interactionPlugin from "@fullcalendar/interaction";
import { v4 as uuid } from "uuid";

function App() {
  const EventItem = ({ info }) => {
    const { event } = info;
    return (
      <div>
        <p>{event.title}</p>
      </div>
    );
  };
  const [events, setEvents] = useState([]);

  const [eventName, setEventName] = useState("");
  const [selectedInfo, setSelectedInfo] = useState({});

  const handleSubmit = () => {
    if (eventName) {
      console.log("events", events);

      setEvents([
        ...events,
        {
          start: selectedInfo.start,
          end: selectedInfo.end,
          title: eventName,
          id: uuid(),
        },
      ]);

      setEventName("");
      setSelectedInfo({});
    }
  };

  return (
    <div className="App">
      <p>Enter your event:</p>
      <input onChange={(e) => setEventName(e.target.value)} value={eventName} />
      <button onClick={handleSubmit}>Submit</button>
      <FullCalendar
        editable
        headerToolbar={{
          start: "today prev next",
          center: "title",
          end: "dayGridMonth dayGridWeek dayGridDay",
        }}
        events={events}
        selectable
        select={setSelectedInfo}
        plugins={[dayGridPlugin, interactionPlugin]}
        views={["dayGridMonth", "dayGridWeek", "dayGridDay"]}
      />
    </div>
  );
}

export default App;
