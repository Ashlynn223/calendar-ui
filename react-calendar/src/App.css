.container {
  width: 100vw;
  height: 100vh;
  background-color: red;
}

.calendar {
  width: 100%;
  height: 400px;
  padding: 20px;
}
