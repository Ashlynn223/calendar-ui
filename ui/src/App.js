import { Calendar, momentLocalizer } from "react-big-calendar";
import moment from "moment";
import "react-big-calendar/lib/css/react-big-calendar.css";
import { useState } from "react";
import s from "./App.module.css";

const localizer = momentLocalizer(moment);
const currentHour = new Date().getHours();
const currentMinute = new Date().getMinutes();
const padNumberWith0 = (num) => num.toString().padStart(2, "0");
const defaultStartTime = `${padNumberWith0(currentHour)}:${padNumberWith0(
  currentMinute,
)}`;

Date.prototype.addHours = (h) => {
  this.setTime(this.getTime() + h * 60 * 60 * 1000);
  return this;
};

const defaultEndTime = `${padNumberWith0(currentHour + 1)}:${padNumberWith0(
  currentMinute,
)}`;

const formatDate = (date) => {
  return [
    date.getFullYear(),
    padNumberWith0(date.getMonth() + 1),
    padNumberWith0(date.getDate()),
  ].join("-");
};

const App = () => {
  const [startDate, setStartDate] = useState(formatDate(new Date()));
  const [endDate, setEndDate] = useState(formatDate(new Date()));
  const [startTime, setStartTime] = useState(defaultStartTime);
  const [endTime, setEndTime] = useState(defaultEndTime);
  const [title, setTitle] = useState("");
  const [events, setEvents] = useState([]);
  const [error, setError] = useState(null);

  const handleSubmit = (e) => {
    e.preventDefault();
    const startDateAndTime = new Date(`${startDate}T${startTime}`);
    const endDateAndTime = new Date(`${endDate}T${endTime}`);

    if (startDateAndTime > endDateAndTime) {
      setError("Error: end cannot be before start.");
      return;
    }

    setEvents([
      {
        title: title,
        start: startDateAndTime,
        end: endDateAndTime,
      },
    ]);

    setError(null);
  };

  return (
    <div className="App">
      <div>
        <Calendar
          localizer={localizer}
          events={events}
          startAccessor="start"
          endAccessor="end"
          style={{ height: 500 }}
        />
      </div>
      <label>Title</label>
      <input
        type="text"
        onChange={(e) => {
          setTitle(e.target.value);
        }}
        value={title}
      />

      <label>Start Date</label>
      <input
        min={formatDate(new Date())}
        type="date"
        onChange={(e) => {
          setStartDate(e.target.value);
        }}
        value={startDate}
      />
      <label>Start Time</label>
      <input
        type="time"
        onChange={(e) => {
          setStartTime(e.target.value);
        }}
        value={startTime}
      />
      <label>End Date</label>
      <input
        min={startDate}
        type="date"
        onChange={(e) => {
          setEndDate(e.target.value);
        }}
        value={endDate}
      />
      <label>End Time</label>
      <input
        type="time"
        onChange={(e) => {
          setEndTime(e.target.value);
        }}
        value={endTime}
      />
      <button onClick={handleSubmit}>Create Event</button>
      {error && <p className={s.error}>{error}</p>}
    </div>
  );
};

export default App;
