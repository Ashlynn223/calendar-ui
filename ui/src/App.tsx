import React, { MouseEvent } from "react";
import moment from "moment";
import { useState } from "react";
import { Calendar, momentLocalizer } from "react-big-calendar";
import "react-big-calendar/lib/css/react-big-calendar.css";
import { CalendarEvent } from "./configs/CalendarEvent";
import s from "./App.module.css";

const App = () => {
  console.log("before all", Date.now());
  const currentHour: number = new Date().getHours();
  const currentMinute: number = new Date().getMinutes();
  const padNumberWith0: Function = (num: Number): string =>
    num.toString().padStart(2, "0");
  const defaultStartTime: string = `${padNumberWith0(
    currentHour,
  )}:${padNumberWith0(currentMinute)}`;
  const defaultEndTime: string = `${padNumberWith0(
    currentHour + 1,
  )}:${padNumberWith0(currentMinute)}`;

  const localizer = momentLocalizer(moment);

  const formatDate: Function = (date: Date): string => {
    console.log("date", date);
    return [
      date.getFullYear(),
      padNumberWith0(date.getMonth() + 1),
      padNumberWith0(date.getDate()),
    ].join("-");
  };

  const [startDate, setStartDate] = useState<string>(formatDate(new Date()));
  const [endDate, setEndDate] = useState<string>(formatDate(new Date()));
  const [startTime, setStartTime] = useState<string>(defaultStartTime);
  const [endTime, setEndTime] = useState<string>(defaultEndTime);
  const [title, setTitle] = useState<string>("");
  const [error, setError] = useState<string | null>(null);
  const [events, setEvents] = useState<CalendarEvent[]>([]);
  console.log({ startDate });

  const handleSubmit = (e: MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    const startDateAndTime: Date = new Date(`${startDate}T${startTime}`);
    const endDateAndTime: Date = new Date(`${endDate}T${endTime}`);

    if (startDateAndTime > endDateAndTime) {
      setError("Error: end cannot be before start.");
      return;
    }

    setEvents([
      {
        title: title,
        start: startDateAndTime,
        end: endDateAndTime,
      },
    ]);
    setTitle("");
    setError(null);
  };

  return (
    <div className="App">
      <div>
        {/* <Calendar
          localizer={localizer}
          events={events}
          startAccessor="start"
          endAccessor="end"
          style={{ height: 500 }}
        /> */}
      </div>
      <label htmlFor="title">Title</label>
      <input
        id="title"
        type="text"
        onChange={(e) => {
          setTitle(e.target.value);
        }}
        value={title}
      />

      <label htmlFor="startDate">Start Date</label>
      <input
        id="startDate"
        min={formatDate(new Date())}
        type="date"
        onChange={(e) => {
          setStartDate(e.target.value);
        }}
        value={startDate}
      />
      <label htmlFor="startTime">Start Time</label>
      <input
        id="startTime"
        type="time"
        onChange={(e) => {
          setStartTime(e.target.value);
        }}
        value={startTime}
      />
      <label htmlFor="endDate">End Date</label>
      <input
        id="endDate"
        min={startDate}
        type="date"
        onChange={(e) => {
          setEndDate(e.target.value);
        }}
        value={endDate}
      />
      <label htmlFor="endTime">End Time</label>
      <input
        id="endTime"
        type="time"
        onChange={(e) => {
          setEndTime(e.target.value);
        }}
        value={endTime}
      />
      <button onClick={handleSubmit}>Create Event</button>
      {error && <p className={s.error}>{error}</p>}
    </div>
  );
};

export default App;
