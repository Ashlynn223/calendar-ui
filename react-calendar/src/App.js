import React, { useState } from "react";
import Calendar from "react-calendar";
import "./Calendar.css";
import "./App.css";

function App() {
  const [value, onChange] = useState(new Date());
  return (
    <div className="container">
      <Calendar className="calendar" onChange={onChange} value={value} />
    </div>
  );
}

export default App;
