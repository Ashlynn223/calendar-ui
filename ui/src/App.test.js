import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App", () => {
  beforeAll(() => {
    jest.useFakeTimers("modern");
    //mock this as utc time zone to stay consistent throughout tests
    jest.setSystemTime(new Date("2017-01-01"));
  });
  afterAll(() => {
    jest.useRealTimers();
  });

  it("renders correctly", () => {
    const app = render(<App />);
    expect(app.asFragment()).toMatchSnapshot();
  });

  it("shows calander", async () => {
    render(<App />);
    expect(await screen.findByText("Month")).toBeVisible();
  });

  describe("events", () => {
    // beforeAll(() => render(<App />)); // this didn't work it may need to be beforeEach instead or something else.
    it("displays correct default values for event inputs on page load", async () => {
      render(<App />);
      expect(await screen.findByLabelText("Start Date")).toHaveValue(
        "2016-12-31"
      );
      expect(await screen.findByLabelText("End Date")).toHaveValue(
        "2016-12-31"
      );
      expect(await screen.findByLabelText("Start Time")).toHaveValue("16:00");
      expect(await screen.findByLabelText("End Time")).toHaveValue("17:00");
    });

    it("displays event in ui when all inputs are provided valid values", async () => {
      render(<App />);
      userEvent.click(await screen.findByLabelText("Title"));
      userEvent.type(
        await screen.findByLabelText("Title"),
        "Berta goes to the baseball game!"
      );
      userEvent.click(screen.getByRole("button", { name: "Create Event" }));
      expect(await screen.findByLabelText("Title")).toHaveAttribute(
        "value",
        ""
      );
      expect(
        screen.getByText("Berta goes to the baseball game!")
      ).toBeVisible();
    });
    xit("errors when end date is before start date", () => {});
    xit("errors when end time is before start time on the same day", () => {});
  });
});
