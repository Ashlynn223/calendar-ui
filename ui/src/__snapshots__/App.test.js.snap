// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div
        class="rbc-calendar"
        style="height: 500px;"
      >
        <div
          class="rbc-toolbar"
        >
          <span
            class="rbc-btn-group"
          >
            <button
              type="button"
            >
              Today
            </button>
            <button
              type="button"
            >
              Back
            </button>
            <button
              type="button"
            >
              Next
            </button>
          </span>
          <span
            class="rbc-toolbar-label"
          >
            December 2016
          </span>
          <span
            class="rbc-btn-group"
          >
            <button
              class="rbc-active"
              type="button"
            >
              Month
            </button>
            <button
              class=""
              type="button"
            >
              Week
            </button>
            <button
              class=""
              type="button"
            >
              Day
            </button>
            <button
              class=""
              type="button"
            >
              Agenda
            </button>
          </span>
        </div>
        <div
          aria-label="Month View"
          class="rbc-month-view"
          role="table"
        >
          <div
            class="rbc-row rbc-month-header"
            role="row"
          >
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Sun
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Mon
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Tue
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Wed
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Thu
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Fri
              </span>
            </div>
            <div
              class="rbc-header"
            >
              <span
                aria-sort="none"
                role="columnheader"
              >
                Sat
              </span>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg rbc-off-range-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    27
                  </button>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    28
                  </button>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    29
                  </button>
                </div>
                <div
                  class="rbc-date-cell rbc-off-range"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    30
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    01
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    02
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    03
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    04
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    05
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    06
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    07
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    08
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    09
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    10
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    11
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    12
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    13
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    14
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    15
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    16
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    17
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    18
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    19
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    20
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    21
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    22
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    23
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    24
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-month-row"
            role="rowgroup"
          >
            <div
              class="rbc-row-bg"
            >
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg"
              />
              <div
                class="rbc-day-bg rbc-today"
              />
            </div>
            <div
              class="rbc-row-content"
              role="row"
            >
              <div
                class="rbc-row "
              >
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    25
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    26
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    27
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    28
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    29
                  </button>
                </div>
                <div
                  class="rbc-date-cell"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    30
                  </button>
                </div>
                <div
                  class="rbc-date-cell rbc-now rbc-current"
                  role="cell"
                >
                  <button
                    class="rbc-button-link"
                    role="cell"
                    type="button"
                  >
                    31
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <label
      for="title"
    >
      Title
    </label>
    <input
      id="title"
      type="text"
      value=""
    />
    <label
      for="startDate"
    >
      Start Date
    </label>
    <input
      id="startDate"
      min="2016-12-31"
      type="date"
      value="2016-12-31"
    />
    <label
      for="startTime"
    >
      Start Time
    </label>
    <input
      id="startTime"
      type="time"
      value="16:00"
    />
    <label
      for="endDate"
    >
      End Date
    </label>
    <input
      id="endDate"
      min="2016-12-31"
      type="date"
      value="2016-12-31"
    />
    <label
      for="endTime"
    >
      End Time
    </label>
    <input
      id="endTime"
      type="time"
      value="17:00"
    />
    <button>
      Create Event
    </button>
  </div>
</DocumentFragment>
`;
